cxx/gendb_test.cc

```diff
@@ -267,35 +268,44 @@ BOOST_AUTO_TEST_CASE(test_get_relation_items) {
 BOOST_AUTO_TEST_CASE(test_unincorporate_reference1) {
   std::mt19937 prng;
   GenDB gendb(&prng, schema);
-  setup_gendb(&prng, gendb);
+  setup_gendb(&prng, gendb, false);
   test_unincorporate_reference_helper(gendb, "Physician", "school", 1, true);
 }
 
 BOOST_AUTO_TEST_CASE(test_unincorporate_reference2) {
   std::mt19937 prng;
   GenDB gendb(&prng, schema);
-  setup_gendb(&prng, gendb);
+  setup_gendb(&prng, gendb, false);
   test_unincorporate_reference_helper(gendb, "Record", "location", 2, true);
 }
 
 BOOST_AUTO_TEST_CASE(test_unincorporate_reference3) {
   std::mt19937 prng;
   GenDB gendb(&prng, schema);
-  setup_gendb(&prng, gendb);
+  setup_gendb(&prng, gendb, false);
+  test_unincorporate_reference_helper(gendb, "Practice", "city", 0, false);
+}
+
+BOOST_AUTO_TEST_CASE(test_unincorporate_reference_new_entities_have_new_parts) {
+  std::mt19937 prng;
+  GenDB gendb(&prng, schema);
+  setup_gendb(&prng, gendb, true);
```

Comment:
I don't think we'll use this with unincorporate_reference, it should just be for initialization of the state. Could you add a test sanity-checking that the state of the gendb object is as expected after incorporating with the flag=true? (e.g. all domain CRP tables should be of size 1, all domain CRPs should have the same number of items incorporated (assuming each domain only appears once in the schema, which it does here)).

Reply:
Added the test you requested, but I believe that all the domain CRP tables should be of size 32, and not 1 (since they are all incorporated into the CRPs with new_id's.)

Reply:
You assert that `tables.size() == 32`, which means that there are 32 tables, each of size 1 (for completeness you could test that each of the tables is indeed size 1, though testing that there are 32 tables and N=32, as you do, is adequate IMO).

Reply:
Added tests that the first tables are of size 1.